e2e/test/ccm_e2e_test.go

```diff
@@ -90,6 +92,14 @@ var _ = Describe("e2e tests", func() {
 		Eventually(watcher.ResultChan()).Should(Receive(EnsuredService()))
 	}
 
+	ensureServiceWasDeleted := func() {
+		notfound := func() bool {
+			_, err := f.LoadBalancer.GetService()
+			return errors.IsNotFound(err)
+		}
+		Eventually(notfound).WithTimeout(10 * time.Second).Should(BeTrue())
```

Comment:
This is impossible to debug, please return the error and check that it's not found, so that we know - in case of failure - if it was nil or a different error than not found.

Reply:
great feedback, this has been refactored